src/Brush.elm

```diff
@@ -0,0 +1,1102 @@
+                                                    , left = resetWithSign (toXSign model.x) (>) .left
+                                                    }
+                                            }
+                                }
+
+                        else
+                            model
+
+                    Nothing ->
+                        model
+
+
+{-| Don't forget the subscriptions, otherwise drag gestures won't work!
+-}
+subscriptions : Brush dim -> (OnBrush -> msg) -> Sub msg
+subscriptions (Brush brush) tagger =
+    Sub.batch
+        [ case brush.drag of
+            Just _ ->
+                Sub.batch
+                    [ Browser.Events.onMouseMove
+                        (D.map
+                            (MouseMove >> tagger)
+                            Events.decodeMousePosition
+                        )
+                    , Browser.Events.onMouseUp (D.succeed (tagger MouseUp))
+                    , Browser.Events.onKeyDown
+                        (D.andThen
+                            (\key ->
+                                case key of
+                                    "Shift" ->
+                                        if brush.x && brush.y then
+                                            D.succeed (tagger ShiftDown)
+
+                                        else
+                                            D.fail ""
+
+                                    "Alt" ->
+                                        D.succeed (tagger AltDown)
+
+                                    -- "Space" ->
+                                    --     D.succeed (tagger SpaceDown)
+                                    _ ->
+                                        D.fail ""
+                            )
+                            (D.field "key" D.string)
+                        )
+                    , Browser.Events.onKeyUp
+                        (D.andThen
+                            (\key ->
+                                case key of
+                                    "Shift" ->
+                                        D.succeed (tagger ShiftUp)
+
+                                    "Alt" ->
+                                        D.succeed (tagger AltUp)
+
+                                    -- "Space" ->
+                                    --     D.succeed (tagger SpaceDown)
+                                    _ ->
+                                        D.fail ""
+                            )
+                            (D.field "key" D.string)
+                        )
+                    ]
+
+            Nothing ->
+                Sub.none
+        ]
+
+
+
+--- View
+
+
+{-| This is a type used to customize the view function of this module. However most of the functions that produce the type
+may appear to also consume it. However, this is not the case, the functions take VirtualDom attributes, but produce this Attribute type.
```

Comment:
Could we have an example of customizing the view function in this way? Is it possible to, say, have the selected area be a circle? Would the handles, etc. still work?

Reply:
A circle would be pretty weird, but possible (although it would be very weird for a 2D brush). I could imagine a (slightly) rounded rect for instance, and certainly a different combination of fill/stroke/filter etc attributes.